xapian-core/weight/tfidfweight.cc

```diff
@@ -36,54 +36,110 @@ using namespace std;
 
 namespace Xapian {
 
-TfIdfWeight::TfIdfWeight(const std::string &normals)
-    : normalizations(normals), param_slope(0.2), param_delta(1.0)
+TfIdfWeight::TfIdfWeight(const std::string& normals)
+    : TfIdfWeight::TfIdfWeight(normals, 0.2, 1.0) {}
```

Comment:
It would be better to just inline this forwarding constructor from the header.

The extra code that adds to the application is very small (I'd expect it's just loading two floating point constants) and it saves having an extra symbol exported by the shared library just for the forwarding version (such symbols need to be resolved at library load time, and that slows start up so it's good to avoid adding symbols for no good reason).

Reply:
Updated